ethcore/src/client/client.rs

```diff
@@ -578,9 +564,49 @@ impl Client {
 		self.db.read().write_buffered(batch);
 		chain.commit();
 		self.update_last_hashes(&parent, hash);
+
+		if let Err(e) = self.prune_ancient(state, &chain) {
```

Comment:
This is still analogous to the prior behavior in the case of 1 entry being pruned: `journal_under` was previously called before `mark_canonical`. Nothing in the `JournalDB` interface (or our implementations) should prohibit "runs" of either call without alternation.